pkg/mesos/types.go

```diff
@@ -0,0 +1,97 @@
+package mesos
+
+import "strings"
+
+// Master represents a single mesos master node
+type Master struct {
+	Host    string `json:"host"`
+	IP      string `json:"ip"`
+	Type    string `json:"type"`
+	Region  string `json:"region"`
+	Zone    string `json:"zone"`
+	ID      string `json:"id"`
+	PID     string `json:"pid"`
+	Version string `json:"version"`
+}
+
+// Resources represents a resource type for a task
+type Resources struct {
+	CPUs  float64 `json:"cpus"`
+	Disk  float64 `json:"disk"`
+	Mem   float64 `json:"mem"`
+	Ports string  `json:"ports"`
+}
+
+// Domain holds information about a nodes region and zone.
+type Domain struct {
+	FaultDomain struct {
+		Region struct {
+			Name string `json:"name"`
+		} `json:"region"`
+		Zone struct {
+			Name string `json:"name"`
+		} `json:"zone"`
+	} `json:"fault_domain"`
+}
+
+// Slave represents a single mesos slave node
+type Slave struct {
+	TaskError           int                    `json:"TASK_ERROR"`
+	TaskFailed          int                    `json:"TASK_FAILED"`
+	TaskFinished        int                    `json:"TASK_FINISHED"`
+	TaskKilled          int                    `json:"TASK_KILLED"`
+	TaskKilling         int                    `json:"TASK_KILLING"`
+	TaskLost            int                    `json:"TASK_LOST"`
+	TaskRunning         int                    `json:"TASK_RUNNING"`
+	TaskStaging         int                    `json:"TASK_STAGING"`
+	TaskStarting        int                    `json:"TASK_STARTING"`
+	TaskUnreachable     int                    `json:"TASK_UNREACHABLE"`
+	Active              bool                   `json:"active"`
+	Attributes          map[string]interface{} `json:"attributes"`
+	Capabilities        []string               `json:"capabilities"`
+	Domain              Domain                 `json:"domain"`
+	FrameworkIDs        []string               `json:"framework_ids"`
+	Hostname            string                 `json:"hostname"`
+	ID                  string                 `json:"id"`
+	PID                 string                 `json:"pid"`
+	Port                int                    `json:"port"`
+	Region              string                 `json:"region"`
+	RegisteredTime      float64                `json:"registered_time"`
+	Resources           Resources              `json:"resources"`
+	UsedResources       Resources              `json:"used_resources"`
+	OfferedResources    Resources              `json:"offered_resources"`
+	ReservedResources   map[string]Resources   `json:"reserved_resources"`
+	UnreservedResources Resources              `json:"unreserved_resources"`
+	Type                string                 `json:"type"`
+	Version             string                 `json:"version"`
+	Zone                string                 `json:"zone"`
+}
+
+// IP returns the IP of a slave parsed from its PID.
+func (s *Slave) IP() string {
+	// PID format: Host@IP:Port
+	s1 := strings.Split(s.PID, "@")
+	s2 := strings.Split(s1[1], ":")
```

Comment:
Is there a simpler way to get slaves IPs?

Reply:
Keeping it for now, to avoid backwards compatibility issues.